pkg/session_recording/recordings.jsx

```diff
@@ -459,10 +495,28 @@
             cockpit.location.go([recording.id], cockpit.location.options);
         }
 
+        componentDidUpdate() {
+            this.drawSortArrow();
+        }
+
         render() {
-            let columnTitles = [_("User"), _("Start"), _("End"), _("Duration")];
+            let columnTitles = [
+                (<div id="user" className="sort" onClick={this.sortBy}><span>{_("User")}</span> <div ref="user"
+                    className="sort-icon"></div></div>),
+                (<div id="start" className="sort" onClick={this.sortBy}><span>{_("Start")}</span> <div ref="start"
+                    className="sort-icon"></div></div>),
+                (<div id="end" className="sort" onClick={this.sortBy}><span>{_("End")}</span> <div ref="end"
+                    className="sort-icon"></div></div>),
+                (<div id="duration" className="sort" onClick={this.sortBy}><span>{_("Duration")}</span> <div
+                    ref="duration" className="sort-icon"></div></div>),
+            ];
             let list = this.props.list;
             let rows = [];
+
+            if (this.state.sorting_field != null) {
+                this.sortList(list);
```

Comment:
I think you are not supposed to modify props, so please make a copy of the list above, when assigning to `list`.

Reply:
Or rather just change the `sortList` function into `getSortedList` and let it deal with checking if there's a sorting field, copying, sorting, and returning the (un)sorted list, and then just assign the result to `list`.